File: backend/app/schemas/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    subject_id: uuid.UUID | None
    original_filename: str
    content_type: str
    file_size_bytes: int
    processing_status: str
    processing_error: str | None
    created_at: datetime
    updated_at: datetime


class DocumentDetail(DocumentRead):
    extracted_text: str | None


class DocumentUpdate(BaseModel):
    subject_id: uuid.UUID | None = None

File: frontend/src/types/document.ts
export type ProcessingStatus = 'pending' | 'processed' | 'failed'
export type IndexingStatus = 'pending' | 'indexed' | 'failed' | 'not_applicable'

export interface Document {
  id: string
  subject_id: string | null
  original_filename: string
  content_type: string
  file_size_bytes: number
  processing_status: ProcessingStatus
  processing_error: string | null
  indexing_status: IndexingStatus
  indexing_error: string | null
  created_at: string
  updated_at: string
}

export interface DocumentDetail extends Document {
  extracted_text: string | null
}

export interface DocumentUpdateRequest {
  subject_id?: string | null
}